pkg/processor/procbuiltin/unwrap.go

```diff
@@ -0,0 +1,350 @@
+	// record must be structured
+	structPayload, ok := rec.Payload.After.(record.StructuredData)
+	if !ok {
+		return record.Record{}, cerrors.Errorf("record payload data must be structured data")
+	}
+
+	// get payload
+	payload := structPayload["payload"]
+	if p, ok := payload.(map[string]interface{}); ok {
+		structPayload = p
+	} else {
+		return record.Record{}, cerrors.Errorf("payload doesn't contain a record")
+	}
+
+	return record.Record{
+		Key:      k.UnwrapKey(rec.Key),
+		Position: rec.Position,
+		Metadata: nil,
+		Payload: record.Change{
+			Before: nil,
+			After:  structPayload,
+		},
+		Operation: record.OperationSnapshot,
+	}, nil
+}
+
+// UnwrapKey unwraps key as a kafka connect formatted record, returns the key's payload content, or returns the
+// original key if payload doesn't exist.
+func (k *KafkaConnectUnwrapper) UnwrapKey(key record.Data) record.Data {
+	// convert the key to structured data
+	var structKey record.StructuredData
+	switch d := key.(type) {
+	case record.RawData:
+		// try unmarshalling raw key
+		err := json.Unmarshal(key.Bytes(), &structKey)
+		// if key is not json formatted, return the original key
+		if err != nil {
+			return key
+		}
+	case record.StructuredData:
+		structKey = d
+	}
+
+	payload, ok := structKey["payload"]
+	// return the original key if it doesn't contain a payload
+	if !ok {
+		return key
+	}
+
+	// if payload is a map, return the payload as structured data
+	if p, ok := payload.(map[string]any); ok {
+		return record.StructuredData(p)
+	}
+
+	// otherwise, convert the payload to string, then return it as raw data
+	raw := fmt.Sprint(payload)
+
+	return record.RawData{Raw: []byte(raw)}
+}
+
+/*
+	 example of debezium record:
+		`{
+		 "payload": {
+		   "after": {
+		     "description": "desc",
+		     "id": 20
+		   },
+		   "before": null,
+		   "op": "c",
+		   "source": {
+		     "opencdc.readAt": "1674061777225877000",
+		     "opencdc.version": "v1",
+		   },
+		   "transaction": null,
+		   "ts_ms": 1674061777225
+		 },
+		 "schema": {} // will be ignored
+		}`
+*/
+// DebeziumUnwrapper unwraps a debezium record from the payload, expects rec.Payload.After to be of type record.StructuredData
+type DebeziumUnwrapper struct{}
+
+const (
+	debeziumOpCreate = "c"
+	debeziumOpUpdate = "u"
+	debeziumOpDelete = "d"
+	debeziumOpRead   = "r" // snapshot
+
+	debeziumFieldBefore    = "before"
+	debeziumFieldAfter     = "after"
+	debeziumFieldSource    = "source"
+	debeziumFieldOp        = "op"
+	debeziumFieldTimestamp = "ts_ms"
+)
+
+func (d *DebeziumUnwrapper) Unwrap(rec record.Record) (record.Record, error) {
+	// record must be structured
+	structPayload, ok := rec.Payload.After.(record.StructuredData)
+	if !ok {
+		return record.Record{}, cerrors.Errorf("record payload data must be structured data")
+	}
+	// get payload
+	payload := structPayload["payload"] // the payload has the debezium record
+	if p, ok := payload.(map[string]interface{}); ok {
+		structPayload = p
+	} else {
```

Comment:
Nitpick: I see indented happy path, I nitpick 🙃 We should generally strive to keep the happy path out of if statements.
```suggestion
	structPayload, ok = structPayload["payload"].(map[string]interface{}) // the payload has the debezium record
	if !ok {
```